### PR TITLE
Add NSP checks to the list of validations

### PR DESCRIPTION
**Pull request by lholmquist**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -192,6 +192,19 @@ function sensitiveDataAudit () {
         });
 }
 
+function runNSPCheck () {
+    var stopSpinner = createSpinner('Performing nsp check');
+    // need to target the local nsp
+    return cmd('./node_modules/.bin/nsp check')
+         .catch(function () {
+            stopSpinner(false);
+            throw "NSP check failed";
```
Comment (inikulin) [inline lib/index.js]: Can we add some more detailed error messages from this thing? Like, what package triggered the fail and so on.

File: lib/index.js [hunk 2]
```diff
@@ -192,6 +192,19 @@ function sensitiveDataAudit () {
...
+        })
+        .then(function (result) {
```
Comment (inikulin) [inline lib/index.js]: `result` is defined but never used

File: lib/index.js [hunk 3]
```diff
@@ -353,7 +369,8 @@ function getOptions (opts) {
         validateGitTag:     true,
         validateBranch:     'master',
         tag:                'latest',
-        prepublishScript:   null
+        prepublishScript:   null,
+        nsp:                true
```
Comment (inikulin) [inline lib/index.js]: Let's name it `nspCheck`

File: package.json [hunk 4]
```diff
@@ -45,14 +46,17 @@
     "log-update": "^1.0.2",
     "node-emoji": "^1.0.4",
     "noop-fn": "^1.0.0",
+    "nsp": "^2.1.0",
     "os-family": "^1.0.0",
     "pinkie-promise": "^1.0.0",
-    "spawn-cmd": "0.0.2"
+    "spawn-cmd": "0.0.2",
+    "uglify-js": "^2.5.0"
```
Comment (inikulin) [inline package.json]: Why we need `ansi2html` and `uglify` deps? I can't they are used in the code.

**Comment by inikulin** [comment]
Thank you for the contribution! But there are some minor issues we need to work through. Also, can you add some basic tests for this feature, please?

File: package.json (hunk 4 above)
Comment (lholmquist) [inline package.json]: right, those shouldn't be there,  they had known vulnerabilities so i was testing them,  forgot to take them out.

**Comment by lholmquist** [comment]
i made the changes from your comments.  I just need to add a couple tests. probably just simple ones similar to the current tests that just check to see if setting/unsetting the option works

**Comment by inikulin** [comment]
Can we add test that it actually failes if vulnerable package is installed. E.g. by running `npm install <pkg>` in the test repo and then trying to publish it?

**Pull request closed without merging**